### ISSUE TITLE
add `createRequire` banner to Function output

### ISSUE THREAD

**Issue by josefaidt** [aws-amplify/amplify-backend#926]
### Environment information

```plain text
System:
    OS: macOS 14.2.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 358.16 MB / 32.00 GB
    Shell: 3.6.4 - /opt/homebrew/bin/fish
  Binaries:
    Node: 20.10.0 - ~/Library/Caches/fnm_multishells/77574_1680319338649/bin/node
    Yarn: 1.22.19 - ~/Library/Caches/fnm_multishells/77574_1680319338649/bin/yarn
    npm: 10.2.3 - ~/Library/Caches/fnm_multishells/77574_1680319338649/bin/npm
    pnpm: 8.14.1 - ~/Library/Caches/fnm_multishells/77574_1680319338649/bin/pnpm
    bun: 1.0.21 - ~/.bun/bin/bun
    Watchman: Not Found
  npmPackages:
    @aws-amplify/backend: ^0.10.1 => 0.10.1 
    @aws-amplify/backend-cli: ^0.9.6 => 0.9.6 
    aws-amplify: ^6.0.12 => 6.0.12 
    aws-cdk: ^2.121.1 => 2.121.1 
    aws-cdk-lib: ^2.121.1 => 2.121.1 
    typescript: ^5 => 5.3.3
```

### Description

see title, enable modifying behavior of esbuild

**Comment by edwardfoyle** [comment]
If we lift all properties of `bundling` into our API we are now tightly coupled to a single implementation. Are there particular properties that we can build an abstraction around to cover the use cases? Also, we could take the stance that if you need bundler props you can use `cdk.NodejsFunction` directly.

**Comment by josefaidt** [comment]
That is a good question, and I am concerned the original issue with esm vs cjs will continue to surface due to `"type": "module"` in the relevant package.json forcing esbuild to build esm bundles. Actually, let me try something..

**Comment by rtpascual** [comment]
defineFunction has the bundling output as ESM to allow for the top level await for executing the function that resolves secrets

**Comment by josefaidt** [comment]
Perhaps we can add this snippet preemptively since folks will need to if using cjs packages
```js
import { createRequire } from "node:module";
const require = createRequire(import.meta.url);
```

**Comment by josefaidt** [comment]
this is now resolved as of [@aws-amplify/backend@0.10.3](https://github.com/aws-amplify/amplify-backend/releases/tag/%40aws-amplify%2Fbackend%400.10.3), closing
